spec/support/shared_context.rb

```diff
@@ -0,0 +1,22 @@
+RSpec.shared_context 'credit card charge submitted' do
+  let(:success) do
+    return true if not block_given?
+    yield
+  end
+  let(:error_message) do
+    return nil if not block_given?
+    yield
+  end
+  before do
+    charge_response = double(:charge)
```

Comment:
You can create test doubles with methods stubbed already so lines 12 and 13 could be refactored like this:

```ruby
charge_response = double(:charge, successful?: success, error_message: error_message)
```